code-challenges.test.js

```diff
@@ -53,4 +89,26 @@ const numbersToAdd3 = []
 // Expected output: []
 
 
+describe("accumulatingSum", () => {
+   it("takes in an array and returns an array of the accumulating sum", () => {
+    const numbersToAdd1 = [2, 4, 45, 9]
+    const numbersToAdd2 = [0, 7, -8, 12]
+    const numbersToAdd3 = []
+     expect(accumulatingSum(numbersToAdd1)).toEqual([2, 6, 51, 60])
+     expect(accumulatingSum(numbersToAdd2)).toEqual([0, 7, -1, 11])
+     expect(accumulatingSum(numbersToAdd3)).toEqual([])
+    })
+})
+
 // b) Create the function that makes the test pass.
+// input accumulatingSum to the array let array sum equal to zero and for the length of the array push array sum. 
+//output accumulatingSum to the array.
+const accumulatingSum = (array) => {
+  let arraySum = 0
+  let accumulatedSumArray = []
+  for (i = 0; i < array.length; i ++) {
+    arraySum = arraySum + array[i]
+    accumulatedSumArray.push(arraySum)
+  }
+    return accumulatedSumArray
+}
```

Comment:
Sean, your code has greatly improved!  These functions and tests are solid!  Please put more into your pseudocode. It will only help you, especially during whiteboarding interviews!  Also,  show us your efforts - your file doesn't have to show only the correct, clean answers.  Show us what you tried.  I know you are putting in a lot of effort - let us see it!  Excellent work!